### PR TITLE
Add ES6 Class support

### PR DESCRIPTION
**Pull request by patricklx**

### CODE REVIEW

File: ember_debug/object-inspector.js [hunk 1]
```diff
@@ -220,7 +220,7 @@ export default EmberObject.extend(PortMixin, {
   },
 
   canSend(val) {
-    return (val instanceof EmberObject) || typeOf(val) === 'array';
+    return val && ((val instanceof Object) || typeOf(val) === 'array');
```
Comment (rwwagner90) [inline ember_debug/object-inspector.js]: Should we check both `Object` and `EmberObject` to account for everything?

Comment (patricklx) [reply ember_debug/object-inspector.js]: done, also pushed an update the better check if `class extends` or `Ember.*.extend` was used by checking if protoype has `_super`

**Comment by rwwagner90** [comment]
@pzuraq @rwjblue do you mind giving this a quick look? We definitely should add support for classes, but I know we have various different flavors of everything, and wanted to make sure we support all the things here, as well as simplify anything we can.

**Comment by rwwagner90** [comment]
@patricklx do you mind fixing the conflicts real quick? Sorry about that!

**Pull request merged**